DEVNOTES.md

```diff
@@ -1,99 +1,119 @@
-# Setting up and building
+* [Prerequisites](#prerequisites)
+* [Building Ignite](#building-ignite)
+* [Running sanity checks](#running-sanity-checks)
+* [Running tests](#running-tests)
+* [Setting up IntelliJ Idea project](#setting-up-intellij-idea-project)
+* [Code structure](#code-structure)
+* [Release candidate verification](#release-candidate-verification)
+
+
 ## Prerequisites
  * Java 11 SDK
  * Maven 3.6.0+ (for building)
- 
-## Building Ignite
-Ignite follows standard guidelines for multi-module maven projects, so it can be easily built using the following 
-command from the project root directory (you can disable the tests when building using `-DskipTests`):
-
-    mvn clean install [-DskipTests]
-
-Upon completion, you will find the CLI tool under `modules/cli/target` directory.
-Use `ignite` on Linux and MacOS, or `ignite.exe` on Windows. 
-
-### Running tests
-To run unit tests only, use the following command:
-
-    mvn test
-
-Before running integration tests, make sure to build the project using the ``package`` target.
-
-To run unit + integration tests, use the following command:
 
-    mvn integration-test
 
-To run integration tests only, use the following command:
-
-    mvn integration-test -Dskip.surefire.tests=true
-    
-### Sanity check targets
-Use the following commands to run generic sanity checks before submitting a PR.
-
-RAT license validation:
-    
-    mvn validate
-    
-Checkstyle code validation:
-
-    mvn checkstyle:checkstyle:aggregate
-
-PMD static code analysis
-
-    mvn compile pmd:check 
+## Building Ignite
+Ignite follows standard guidelines for multi-module maven projects, so it can be easily built using the following command from the project root directory (you can disable the tests when building using `-DskipTests`):
+```
+mvn clean package [-DskipTests]
+```
+Upon build completion, CLI tool can be found be under `modules/cli/target` directory. Use `ignite` on Linux and MacOS, or `ignite.exe` on Windows.
+
+
+## Running sanity checks
+### Code style
+Code style is checked with [Apache Maven Checkstyle Plugin](https://maven.apache.org/plugins/maven-checkstyle-plugin/).
+
+[Checkstyle rules](check-rules/checkstyle-rules.xml).
+
+[Checkstyle suppressions](check-rules/checkstyle-suppressions.xml).
+```
+mvn clean checkstyle:checkstyle-aggregate
+```
+Code style check result is generated at `target/site/checkstyle-aggregate.html`
+
+### License headers
+Project files license headers match with required template is checked with [Apache Rat Maven Plugin](https://creadur.apache.org/rat/apache-rat-plugin/).
+```
+mvn clean apache-rat:check -pl :apache-ignite
```

Comment:
Execute `mvn clean apache-rat:check -pl :apache-ignite` from the project root.

Result FAILED:

```
[ERROR] Failed to execute goal org.apache.rat:apache-rat-plugin:0.13:check (default-cli) on project apache-ignite: Too many files with unapproved license: 12 See RAT report in: C:\git\ignite-3\target\rat.txt -> [Help 1]
```

`rat.txt` snippet:

```
Files with unapproved licenses:

  modules/cli/ignite-cli.iml
  modules/cli-common/ignite-cli-common.iml
  modules/configuration/ignite-configuration.iml
  modules/configuration-annotation-processor/ignite-configuration-annotation-processor.iml
  modules/metastorage-client/metastorage-client.iml
  modules/metastorage-common/metastorage-common.iml
  modules/network/ignite-network.iml
  modules/raft/ignite-jraft.iml
  modules/raft/raft.iml
  modules/raft-client/ignite-raft-client.iml
  modules/rest/ignite-rest.iml
  modules/runner/ignite-runner.iml
```

So `*.iml` files should be excluded from target files.

Reply:
Added  exclusion, please check.